### PR TITLE
Improve ENS Address Input

### PR DESCRIPTION
**Pull request by industral**
https://www.dropbox.com/s/160lnm6f21xqht9/metamask.mov?dl=0

### CODE REVIEW

File: ui/app/components/app/ens-input.js [hunk 1]
```diff
@@ -79,19 +83,21 @@ EnsInput.prototype.componentDidMount = function () {
 
 EnsInput.prototype.lookupEnsName = function (recipient) {
   const { ensResolution } = this.state
+  recipient = recipient.trim()
 
   log.info(`ENS attempting to resolve name: ${recipient}`)
-  this.ens.lookup(recipient.trim())
+  this.ens.lookup(recipient)
```
Comment (tmashuang) [inline ui/app/components/app/ens-input.js]: We use `trim` here to remove the white space from copy-pastes

Comment (industral) [reply ui/app/components/app/ens-input.js]: I move it above. please see line 86

**Review by tmashuang — APPROVED** [review]
LGTM

**Comment by bdresser** [comment]
hey @industral @tmashuang we're reverting this in https://github.com/MetaMask/metamask-extension/pull/6691 - it introduces some buggy behavior in the TO field (video below)

**Comment by danfinlay** [comment]
@bdresser Video did not attach.

**Comment by danfinlay** [comment]
Nevermind! It did, I guess it was added in an edit.

**Comment by tmashuang** [comment]
As the video suggests, there's issues with changing the network while having an address or ens name. Different issues arise when going from Mainnet -> Testnet and vice-versa.

File: ui/app/pages/send/to-autocomplete/to-autocomplete.js [hunk 2]
```diff
@@ -93,24 +94,34 @@ ToAutoComplete.prototype.componentDidUpdate = function (nextProps) {
...
+
+  return h('div.send-v2__to-autocomplete', {style: {
+      borderColor: inError ? 'red' : null,
+    }}, [
 
     h(`input.send-v2__to-autocomplete__input${qrScanner ? '.with-qr' : ''}`, {
       placeholder: this.context.t('recipientAddress'),
       className: inError ? `send-v2__error-border` : '',
-      value: to,
+      value: recipient,
```
Comment (danjm) [inline ui/app/pages/send/to-autocomplete/to-autocomplete.js]: Maybe `recipient || to` instead?

Comment (danjm) [reply ui/app/pages/send/to-autocomplete/to-autocomplete.js]: @tmashuang Did you happen to try this out with this branch/changeset?

**Comment by danjm** [comment]
I believe the issue shown in the video https://cl.ly/8fb6fb8c9555, and the motivation for the revert, was an issue before this PR / before the reverted commit. There are other small bugs as Thomas mentioned. My current belief is that this PR introduced no new bugs.

**Pull request merged**